server/graphql/general/src/mutations/barcode.rs

```diff
@@ -0,0 +1,81 @@
+use async_graphql::*;
+
+use graphql_core::{
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use graphql_types::types::BarcodeNode;
+use repository::Barcode;
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    barcode::InsertBarcodeError as ServiceError,
+};
+#[derive(InputObject)]
+#[graphql(name = "InsertBarcodeInput")]
+pub struct BarcodeInput {
+    pub value: String,
+    pub item_id: String,
+    pub pack_size: Option<i32>,
+}
+
+#[derive(Union)]
+#[graphql(name = "InsertBarcodeResponse")]
+pub enum InsertResponse {
+    Response(BarcodeNode),
+}
+
+#[derive(SimpleObject)]
+pub struct InsertBarcodeError {
+    pub error: String,
+}
+
+impl BarcodeInput {
+    pub fn to_domain(&self) -> service::barcode::BarcodeInput {
```

Comment:
usually to_domain will `own` self, in which case there is no need to `clone()`

Reply:
cool, thanks